pkg/controller/nodemaintenance/nodemaintenance_controller_test.go

```diff
@@ -249,3 +260,335 @@ var _ = Describe("updateCondition", func() {
 
 	})
 })
+
+type FakeDiscovery struct {
+	k8sdiscovery.DiscoveryInterface
+	groupList *metav1.APIGroupList
+}
+
+func (self *FakeDiscovery) ServerGroups() (*metav1.APIGroupList, error) {
+	return self.groupList, nil
+}
+
+type FakeClient struct {
+	k8sfakeclient.Clientset
+	fakeDiscovery *FakeDiscovery
+}
+
+func (self *FakeClient) Discovery() k8sdiscovery.DiscoveryInterface {
+	return self.fakeDiscovery
+}
+
+var _ = Describe("checkLeaseSupport", func() {
+	var fcd *FakeClient
+	var cl client.Client
+
+	setFakeClients := func(withLease bool) {
+
+		cl = fake.NewFakeClient()
+		if withLease {
+			fcd = &FakeClient{fakeDiscovery: &FakeDiscovery{groupList: &metav1.APIGroupList{
+				Groups: []metav1.APIGroup{{
+					Name: "lease",
+					Versions: []metav1.GroupVersionForDiscovery{{
+						GroupVersion: LeaseApiPackage,
+						Version:      "v1beta1",
+					}},
+				}},
+			}}}
+		} else {
+		   fcd = &FakeClient{fakeDiscovery: &FakeDiscovery{groupList: &metav1.APIGroupList{
+				Groups: []metav1.APIGroup{{
+					Name: "lease",
+					Versions: []metav1.GroupVersionForDiscovery{{
+						GroupVersion: LeaseApiPackage + "no",
+						Version:      "v1beta1",
+					}},
+				}},
+			}}}
+		}
+	}
+
+
+	Context("check for leases test", func() {
+
+		It("check lease supported", func() {
+
+				setFakeClients(true)
+
+				// Create a ReconcileNodeMaintenance object with the scheme and fake client
+				r := &ReconcileNodeMaintenance{client: cl}
+				err := initDrainer(r, fcd)
+
+				Expect(err).NotTo(HaveOccurred())
+				Expect(r.isLeaseSupported).To(Equal(true))
+
+		})
+
+		It("check lease not supported", func() {
+
+				setFakeClients(false)
+
+				// Create a ReconcileNodeMaintenance object with the scheme and fake client
+				r := &ReconcileNodeMaintenance{client: cl}
+				err := initDrainer(r, fcd)
+
+				Expect(err).NotTo(HaveOccurred())
+				Expect(r.isLeaseSupported).To(Equal(false))
+
+		})
+	})
```

Comment:
This block needs to look like: https://github.com/kubernetes/dashboard/blob/master/src/app/backend/validation/validaterbacstatus_test.go#L105-L121

Specifically, a loop that runs the same code for each case and operates on an array of cases that include a description, the test input and all expected output(s).

Reply:
done